specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,1204 @@
+          "items": {
+            "type": "string"
+          }
+        },
+        "approvedPlans": {
+          "description": "Gets or sets Approved plans ids, empty in case of rejected",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "comment": {
+          "description": "Gets or sets admin comment",
+          "type": "string"
+        },
+        "user": {
+          "description": "Gets or sets admin details",
+          "type": "string"
+        }
+      }
+    },
+    "AdminGetApprovalRequestsResponse": {
+      "description": "Admin action properties data structure",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Gets or sets Id",
+          "type": "string"
+        },
+        "publisherId": {
+          "description": "Gets or sets publisher Id",
+          "type": "string"
+        },
+        "offerId": {
+          "description": "Gets or sets offer Id",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Gets or sets display name",
+          "type": "string"
+        },
+        "plans": {
+          "description": "Gets or sets plans details",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PlanRequestsDetails"
+          }
+        },
+        "messageCode": {
+          "description": "Gets or sets the notification message id",
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "PrivateStoreNotificationsList": {
+      "description": "Describes the json payload for the list of notifications",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateStoreNotificationsResource"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next set of notifications list results if there are any.",
+          "type": "string"
+        }
+      }
+    },
+    "PrivateStoreNotificationsResource": {
+      "description": "The privateStore notifications data structure.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateStoreNotificationsProperties",
+          "description": "The privateStore notifications data structure.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "PrivateStoreNotificationsProperties": {
+      "description": "Notifications data structure properties",
+      "type": "object",
+      "properties": {
+        "details": {
+          "description": "Notifications details",
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/PrivateStoreNotifications"
+          },
+          "readOnly": true
+        },
+        "update": {
```

Comment:
I commented on this in the first round.  "This body has special properties used for the update that affect other properties within the resulting resource? That does not follow API guidelines. The PUT should be directly manipulating the resulting properties. If that is not possible for some reason then this needs to be modeled as a POST action".  You have a special "update" section of this resource that is only used for PUTs and supposedly changes properties in the details section. Properties should be changed directly by the user, not proxied through another part of the resources payload.  If that is not possible for some types of updates (should be rare) then the update needs to happen via a POST action.

Reply:
All of the resources in this PR have that same problem.  Resources should not have a "payload" and "response" section within them.  The resource is the payload and the response itself.

Reply:
We cleared out the resources according to the RPC documentation